### PR TITLE
Added functions for printing EVP_PKEYs to FILE *

### LINKED ISSUE #14172: print key to file stream
All "elementary" keys support " human-readable " output to file stream. All those keys are deprecated in 3.0:
```
$ grep print_fp util/libcrypto.num 
DSAparams_print_fp                      229     3_0_0   EXIST::FUNCTION:DEPRECATEDIN_3_0,DSA,STDIO
DSA_print_fp                            278     3_0_0   EXIST::FUNCTION:DEPRECATEDIN_3_0,DSA,STDIO
ECPKParameters_print_fp                 816     3_0_0   EXIST::FUNCTION:DEPRECATEDIN_3_0,EC,STDIO
X509_REQ_print_fp                       1110    3_0_0   EXIST::FUNCTION:STDIO
RSA_print_fp                            1119    3_0_0   EXIST::FUNCTION:DEPRECATEDIN_3_0,STDIO
EC_KEY_print_fp                         1121    3_0_0   EXIST::FUNCTION:DEPRECATEDIN_3_0,EC,STDIO
DHparams_print_fp                       2111    3_0_0   EXIST::FUNCTION:DEPRECATEDIN_3_0,DH,STDIO
X509_print_fp                           2463    3_0_0   EXIST::FUNCTION:STDIO
X509_CRL_print_fp                       2745    3_0_0   EXIST::FUNCTION:STDIO
X509V3_EXT_print_fp                     2815    3_0_0   EXIST::FUNCTION:STDIO
ECParameters_print_fp                   3318    3_0_0   EXIST::FUNCTION:DEPRECATEDIN_3_0,EC,STDIO
BN_print_fp                             3625    3_0_0   EXIST::FUNCTION:STDIO
```

Since 1.0 EVP_PKEY supports output using BIO functionality but lack direct output to files:
```
$ grep PKEY.*print util/libcrypto.num 
EVP_PKEY_print_public                   1919    3_0_0   EXIST::FUNCTION:
EVP_PKEY_print_params                   2795    3_0_0   EXIST::FUNCTION:
EVP_PKEY_print_private                  2970    3_0_0   EXIST::FUNCTION:
```

This is an enhancement request to add functions that outputs directly to files.

Use case:  Now PKIX-SSH is rewritten to use pkey for identity and key exchange. In some operations it still uses "elementary" key. One case is for "debugging" purposes and would like to eliminate custom source here.

### PR DESCRIPTION
**Pull request by t8m**
Fixes #14172

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

**Comment by t8m** [comment]
The sanitizers issue is unrelated and fixed by PR #14573

File: crypto/evp/p_lib.c [hunk 1]
```diff
@@ -981,17 +978,14 @@ const char *evp_pkey_type2name(int type)
...
 {
-#ifndef FIPS_MODULE
     if (pkey->keymgmt == NULL) {
         int type = evp_pkey_name2type(name);
 
         return pkey->type == type;
     }
-#endif
     return EVP_KEYMGMT_is_a(pkey->keymgmt, name);
 }
```
Comment (levitte) [inline crypto/evp/p_lib.c:990]: Looks like you made `EVP_PKEY_is_a()` unavailable in the FIPS provider...  why?

**Comment by t8m** [comment]
Not really. It was already inside ifndef FIPS_MODULE. There were two same ifndefs nested.

16. 3. 2021 19:38, 19:38, Richard Levitte ***@***.***> napsal/a:
***@***.*** commented on this pull request.

**Comment by levitte** [comment]
> Not really. It was already inside ifndef FIPS_MODULE. There were two same ifndefs nested.

Yeah, that begs an investigation...  that might be the result of a merge gone awry...  Essentially, there's *no* reason why `EVP_PKEY_is_a()` shouldn't be accessible from the FIPS module.

**Comment by t8m** [comment]
Apparently it is not used there otherwise I'd expect getting build failures when that outer ifndef was added. What would you actually use that function for inside a provider? Do we even use EVP_PKEYs inside providers? Perhaps inside encoders/decoders but they aren't in the fips provider.

16. 3. 2021 20:33, 20:33, Richard Levitte ***@***.***> napsal/a:

**Comment by levitte** [comment]
> Apparently it is not used there otherwise I'd expect getting build failures when that outer ifndef was added. What would you actually use that function for inside a provider? Do we even use EVP_PKEYs inside providers? Perhaps inside encoders/decoders but they aren't in the fips provider.

The FIPS provider is a bit special in so far that it has an internal provider,  so even though EVP_PKEY_is_a() isn't *currently* used there, it may potentially be at some point.

**Comment by paulidale** [comment]
@levitte, the flip side is anything we include in the FIPS provider becomes less mutable.  Let's add it when it is needed rather than because it might be in the future.

File: doc/man3/EVP_PKEY_print_private.pod [hunk 2]
```diff
@@ -2,33 +2,45 @@
...
+                              int indent, ASN1_PCTX *pctx);
 
 =head1 DESCRIPTION
 
 The functions EVP_PKEY_print_public(), EVP_PKEY_print_private() and
 EVP_PKEY_print_params() print out the public, private or parameter components
-of key B<pkey> respectively. The key is sent to BIO B<out> in human readable
-form. The parameter B<indent> indicated how far the printout should be indented.
+of key I<pkey> respectively. The key is sent to B<BIO> I<out> in human readable
+form. The parameter I<indent> indicated how far the printout should be indented.
```
Comment (slontis) [inline doc/man3/EVP_PKEY_print_private.pod:31]: indicates

Comment (t8m) [reply doc/man3/EVP_PKEY_print_private.pod:31]: Fixed.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
AppVeyor failure is unrelated.

**Comment by t8m** [comment]
Merged to master. Thank you for the reviews.

**Pull request closed without merging**